neuron_morphology/transforms/tilt_correction/compute_tilt_correction.py

```diff
@@ -0,0 +1,226 @@
+
+        Parameters
+        ----------
+        morphology: Morphology object
+        soma_voxel: soma voxel in ccf {'x': , 'y': , 'z': }
+        slice_angle_matrix: 4 x 4 affine matrix of the slice plane
+            relative to ccf
+        closest_path: 3 x N array of voxel coordinates in closest streamline
+
+        Returns
+        -------
+        tilt angle correction (radians)
+
+    """
+
+    # Find slice plane vector
+    M = slice_angle_matrix[0:3, :]
+    a2 = np.dot(M, np.array([0, 0, 0, 1]))
+    b2 = np.dot(M, np.array([1, 0, 0, 1]))
+    c2 = np.dot(M, np.array([0, 1, 0, 1]))
+
+    norm_vec = np.cross(b2 - a2, c2 - a2)
+    norm_unit = norm_vec / euclidean(norm_vec, [0, 0, 0])
+
+    # Find approximate streamline vector
+    if euclidean(closest_path[:, 0], soma_voxel) == 0:
+        # if soma is at the top, calculate from other end of streamline
+        streamline_unit = (soma_voxel - closest_path[:, -1]) /\
+            euclidean(soma_voxel, closest_path[:, -1])
+    else:
+        streamline_unit = (closest_path[:, 0] - soma_voxel) /\
+             euclidean(closest_path[:, 0], soma_voxel)
+
+    # determine angle between
+    s = np.linalg.norm(np.cross(norm_unit, streamline_unit))
+    dp = np.dot(norm_unit, streamline_unit)
+    theta = np.arctan2(s, dp)
+
+    tilt_angle = np.pi / 2 - theta
+
+    return tilt_angle
+
+
+def load_path_ids_and_voxels(ccf_path):
+    vi = h5py.File(ccf_path, 'r')
+
+    # initialize
+    path_lookup_table = vi['view lookup'][:][np.where(vi['view lookup'][:] > -1)]
+    paths = vi['paths'][:]
+    vi.close()
+
+    voxel_idxs = []
+    path_id_from_voxel_idx = {}
+
+    for path_id in path_lookup_table:
+        path = paths[path_id, :]
+        path = path[path > 0]
+        for voxel in path:
+            path_id_from_voxel_idx[voxel] = [path_id]
+        voxel_idxs.extend(path)
+
+    return paths, path_id_from_voxel_idx, voxel_idxs
+
+
+def find_closest_path_id(soma_voxel: Dict[str, int],
+                         path_id_from_voxel_idx: Dict[int, int],
+                         voxel_idxs: List[int]):
+
+    voxels = np.array(np.unravel_index(voxel_idxs, CCF_SHAPE))
+    delta = voxels - np.reshape(soma_voxel, (3, 1))
+    distances = np.linalg.norm(delta, axis=0)
+    min_voxel = voxels[:, distances.argmin()]
+    min_voxel_idx = np.ravel_multi_index(min_voxel, CCF_SHAPE)
+    closest_path_id = path_id_from_voxel_idx[min_voxel_idx]
+
+    return closest_path_id
+
+
+def determine_slice_flip(morphology: Morphology,
+                         soma_marker: Dict,
+                         slice_image_flip: bool):
+    """
+        Determines whether the tilt correction should be positive or negative
+
+        Parameters
+        ----------
+        morphology: Morphology object
+        soma_marker: soma marker dictionary from reconstruction marker file
+        slice_image_flip: indicates whether the image was flipped relative
```

Comment:
In which sense image could be flipped? left to right, up vs down? Please elaborate